plugins/base/src/main/resources/dokka/scripts/navigation-loader.js

```diff
@@ -25,6 +29,150 @@ displayNavigationFromPage = () => {
             });
         });
     });
+    window.addEventListener('popstate', (event) => {
+      loadPage(document.location, false)
+    });
+}
+
+const selectPage = (event, elem) => {
+    const link = elem.getAttribute("href")
+    const anchorOrAbsoluteUrlRegex = /^#|^(?:[a-z]+:)?\/\//gi;
+    if(link.match(anchorOrAbsoluteUrlRegex))
+        return;
+     event.preventDefault();
+     event.stopPropagation();
+     event.stopImmediatePropagation();
+     loadPage(link, true)
+}
+
+const loadPage = (link, isPushState = true) => {
+    navigationPageText = fetch(link).then(response => response.text()).then(data => {
+        var parser = new DOMParser();
+        var doc = parser.parseFromString(data, "text/html");
+        addBase(doc, link)
+
+        const oldPathToRoot = pathToRoot
+        updatePathToRoot(doc)
+        replacePathToRoot(oldPathToRoot)
+        convertLinksToAbsoluteInHead()
+
+        document.getElementById("main").innerHTML = doc.getElementById("main").innerHTML;
+        document.getElementById("navigation-wrapper").innerHTML = doc.getElementById("navigation-wrapper").innerHTML;
+        document.getElementsByTagName("title")[0].innerHTML = doc.getElementsByTagName("title")[0].innerHTML;
+
+        if(isPushState) window.history.pushState({}, '', link);
+
+        document.querySelectorAll("#main a")
+          .forEach(elem => elem.addEventListener('click', (event) => selectPage(event, elem)))
```

Comment:
I guess these event listeners were initialized before the comment about event delegation was written by me.